src/commonControls/ValueControl.ts

```diff
@@ -120,6 +120,12 @@ export interface ValueControlProps extends ControlProps {
      * Props to configure input handling.
      */
     inputHandling?: ControlInputHandlingProps;
+
+    /**
+     * Function that maps the slot value to a corresponding rendered value
+     * that will be presented to the user.
+     */
+    valueRenderer?: (slotValueId: string, input: ControlInput) => string;
```

Comment:
Fixed.

Reply:
After seeing your updates, I think 'value' or 'itemId' is the good choices rather than (slotValue.. control state shouldn't talk about slots).

For other controls, they can use 'value' or the name of the sub-property if there is one, e.g. 'startDate' and so on.